api/src/domain/license-status/searchBusinessAddressesForMatch.test.ts

```diff
@@ -0,0 +1,107 @@
+import { generateLicenseSearchAddress } from "@shared/test";
+import { searchBusinessAddressesForMatch } from "./searchBusinessAddressesForMatch";
+
+describe("searchBusinessAddressesForMatch", () => {
+  const userInputtedAddress = generateLicenseSearchAddress({
+    addressLine1: "Main St",
+    addressLine2: "Apt 1",
+    zipCode: "12345",
+  });
+  const matchedAddress1 = generateLicenseSearchAddress({
+    addressLine1: "Main St",
+    addressLine2: "Apt 1",
+    zipCode: "12345",
+  });
+  const matchedAddress2 = generateLicenseSearchAddress({
+    addressLine1: "Main St",
+    addressLine2: "Apt 1",
+    zipCode: "12345",
+  });
+  const randomBusinessWithRandomAddresses1 = {
+    businessId1234: [generateLicenseSearchAddress({}), generateLicenseSearchAddress({})],
+  };
+  const randomBusinessWithRandomAddresses2 = {
+    businessId2345: [generateLicenseSearchAddress({}), generateLicenseSearchAddress({})],
+  };
+  const businessWithNoAddresses = {
+    businessId3456: [],
+  };
+
+  describe("single business ids", () => {
+    it("returns correct business id when it has two matching addresses", () => {
+      const businessId1 = { matchingId: [matchedAddress1, matchedAddress2] };
+      const result = searchBusinessAddressesForMatch([businessId1], userInputtedAddress);
+      expect(result).toEqual("matchingId");
+    });
+
+    it("throws error when there is no match", () => {
+      expect(() =>
+        searchBusinessAddressesForMatch([randomBusinessWithRandomAddresses1], userInputtedAddress)
+      ).toThrow("NO_MATCH");
+    });
+
+    it("throws error when there are no addresses", () => {
+      expect(() => searchBusinessAddressesForMatch([businessWithNoAddresses], userInputtedAddress)).toThrow(
+        "NO_MATCH"
+      );
+    });
+  });
+
+  describe("multiple business ids", () => {
+    it("returns correct business id when it has one matching addresses", () => {
+      const businessIdWithMatchingAddress = {
+        matchingId: [matchedAddress1, generateLicenseSearchAddress({})],
+      };
+      const result = searchBusinessAddressesForMatch(
+        [businessIdWithMatchingAddress, randomBusinessWithRandomAddresses1],
+        userInputtedAddress
+      );
+      expect(result).toEqual("matchingId");
+    });
+
+    it("returns correct business id when it has two matching addresses", () => {
+      const businessIdWithMatchingAddress = { matchingId: [matchedAddress1, matchedAddress2] };
+      const result = searchBusinessAddressesForMatch(
+        [businessIdWithMatchingAddress, randomBusinessWithRandomAddresses1],
+        userInputtedAddress
+      );
+      expect(result).toEqual("matchingId");
+    });
+
+    it("returns correct business id when it has one matching addresses and other has no address", () => {
+      const businessIdWithMatchingAddress = {
+        matchingId: [matchedAddress1, generateLicenseSearchAddress({})],
```

Comment:
[dust/sand] This occurred to me as I'm reading through the test and code here. Is it possible to ever have one of these `BusinessIdToAddressesMapping` types with more than one ID key? For example:
```
  const businessIdWithMatchingAddress = {
        "matchingId": [matchedAddress1, generateLicenseSearchAddress({})],
        "some-other-id" [adddress1]
  }
```

If so: I think we need a test for that case.

If not: Then I don't think a record is the appropriate type to represent this. If there's only ever ONE id with addresses, then why have a record (that you do Object.entries on)? A more appropriate type might be:
```
type X = {
  businessId: string;
  addresses: LicenseSearchAddress[]
}
```